src/modules/handlers/ManifestHandler.js

```diff
@@ -19,40 +19,88 @@ DB.version(1).stores({
+      DB.table("DestinyInventoryItemLiteDefinition").toCollection().first(),
+      DB.table("DestinyObjectiveDefinition").toCollection().first(),
+      DB.table("DestinyProgressionDefinition").toCollection().first(),
+      DB.table("DestinyTalentGridDefinition").toCollection().first(),
+      DB.table("DestinyVendorDefinition").toCollection().first()
+    ]);
+
     //Add data to global manifest object
     MANIFEST = {
-      "DestinyActivityDefinition": values[0].data,
-      "DestinyActivityTypeDefinition": values[1].data,
-      "DestinyActivityModeDefinition": values[2].data,
-      "DestinyCollectibleDefinition": values[3].data,
-      "DestinyPresentationNodeDefinition": values[4].data,
-      "DestinyRecordDefinition": values[5].data,
-      "DestinyInventoryItemDefinition": values[6].data,
-      "DestinyInventoryItemLiteDefinition": values[6].data,
-      "DestinyObjectiveDefinition": values[7].data,
-      "DestinyProgressionDefinition": values[8].data,
-      "DestinyTalentGridDefinition": values[9].data,
-      "DestinyVendorDefinition": values[10].data
+      DestinyActivityDefinition: values[0].data,
+      DestinyActivityTypeDefinition: values[1].data,
+      DestinyActivityModeDefinition: values[2].data,
+      DestinyCollectibleDefinition: values[3].data,
+      DestinyPresentationNodeDefinition: values[4].data,
+      DestinyRecordDefinition: values[5].data,
+      DestinyInventoryItemDefinition: values[6].data,
+      DestinyInventoryItemLiteDefinition: values[6].data,
+      DestinyObjectiveDefinition: values[7].data,
+      DestinyProgressionDefinition: values[8].data,
+      DestinyTalentGridDefinition: values[9].data,
+      DestinyVendorDefinition: values[10].data
     };
-  }).catch((error) => { callback({ status: 'failedManifestMount', statusText: "Failed to Mount Manifest, Try Refresh?", error: true, loading: true, manifestMounted: false }); console.log("Error occured trying to grab manifest from IndexDB"); });
-  console.log("Manifest Unpacked Successfully");
-  callback({ status: 'ready', statusText: '', loading: false, manifestMounted: true });
+
+    // Check if the manifest components are empty
+    Object.keys(MANIFEST).forEach((key) => {
+      if (MANIFEST[key].length === 0) {
+        isManifestRefreshRequired = true;
+        console.warn(`Manifest is corrupted ('${key}' is empty)`);
+      }
+    });
+
+    // If data is enpty, refresh from Bungie
+    if (isManifestRefreshRequired) {
+      await ClearManifest();
+      await Load(callback);
+    } else {
+      console.log("Manifest Unpacked Successfully");
+      callback({
+        status: "ready",
+        statusText: "",
+        loading: false,
+        manifestMounted: true
+      });
+    }
+  } catch (err) {
+    callback({
+      status: "failedManifestMount",
+      statusText: "Failed to Mount Manifest, Try Refresh?",
+      error: true,
+      loading: true,
+      manifestMounted: false
+    });
+    DB.clear();
```

Comment:
`DB.clear()` does not work due to DB not having a clear function, you'd need to do it for each child database, so a good call to do here instead would be `ClearManifest()`

Reply:
I honestly dont even remember adding this one. Might have autocompleted something by mistake, sorry!

Reply:
No need to be sorry! Just something I found whilst testing. It's good code!